packages/grid/x-data-grid/src/components/cell/GridEditBooleanCell.tsx

```diff
@@ -59,12 +72,17 @@ export function GridEditBooleanCell(
   const classes = useUtilityClasses(ownerState);
 
   const handleChange = React.useCallback(
-    (event: React.ChangeEvent<HTMLInputElement>) => {
+    async (event: React.ChangeEvent<HTMLInputElement>) => {
       const newValue = event.target.checked;
+
+      if (onChange) {
```

Comment:
Should we also call `onChange` in our code example of *Create your own edit component* ?

If not, generic custom edit components done by the user following our doc will not support the use case fixed by this PR
But it makes the example a little heavier.

Reply:
The purpose of adding `onChange` is to allow to do something when the value is changed in the built-in components, where the user can't change the code that's running. In the example of _Create your own edit component_ it's the opposite, the user can change the code. I don't know if it's worth to add `onChange` to it, since at this point of the page the user is only learning how to create custom components.